xarray/core/dataset.py

```diff
@@ -3803,7 +3837,38 @@ def unstack(
 
         result = self.copy(deep=False)
         for dim in dims:
-            result = result._unstack_once(dim, fill_value, sparse)
+
+            if (
+                # Dask arrays don't support assignment by index, which the fast unstack
+                # function requires.
+                # https://github.com/pydata/xarray/pull/4746#issuecomment-753282125
+                any(is_duck_dask_array(v.data) for v in self.variables.values())
+                # Sparse doesn't currently support (though we could special-case
+                # it)
+                # https://github.com/pydata/sparse/issues/422
+                or any(
+                    isinstance(v.data, sparse_array_type)
```

Comment:
Looks like this will work on `sparse` master branch, shall we change this to a LooseVersion check too?

Reply:
Yes in principle, but I think waiting for a version we can actually test against might be safer.

Because of the pint issue, this also is only symbolic at this stage — currently we only pass numpy arrays because we can't check pint types directly. (though also could work around this easily by checking for numpy or sparse types)

Reply:
:+1: OK we can sort it out later